xboxlib.c

```diff
@@ -184,15 +184,21 @@ int read_rsafrombin_asterix()
 	return 0;
 }    
 
-void gigimport(giant g, unsigned char *buff, int len) {
+void gigimport(giant g, const unsigned char *buff, int len) {
 
 	// copy buffered 'number' into giant's number buffer
 	memcpy(g->n, buff, len);
 
 	assert((len % 2) == 0);
 
+	// Get number of shorts
 	g->sign = len / 2;
 
+	// Only count used shorts
+	while((g->sign >= 1) && (g->n[g->sign - 1] == 0)) {
+		g->sign -= 1;
+	}
+
 	assert(g->sign != 0);
```

Comment:
This assert gets triggered for nxdk XBEs now, because `buff` is `len` bytes of zero.

I'm not sure why this check exists in the first place?

Reply:
This seems to be a hack that was introduced here: https://github.com/XboxDev/xbedump/commit/dd703113878c8ae814c71c99aee3af640447c619#diff-e64a6c66ddd61d0fa1f0670ca3423da2L227

---

I don't think the additional check / fixup for `g->sign == 0` isn't even necessary, because https://github.com/XboxDev/xbedump/blob/d0d6508db7a4e5fcf159dab40853c695b4c52e6e/giants.c#L359-L377 also checks if the length is zero. So `g->sign == 0` appears to be a valid state.

It also handles 0 like that in this:

https://github.com/XboxDev/xbedump/blob/d0d6508db7a4e5fcf159dab40853c695b4c52e6e/giants.c#L393-L407

Reply:
I've removed this assert.